Add an API for LLDB to check whether a function throws

adrian-prantl opened the pull request:
Add an API for LLDB to check whether a function throws
based on its mangled name.

When stepping out of a function using the Swift calling convention,
the debugger will need to know whether to check the error return
register for an exception or not.

<rdar://problem/29481673>

adrian-prantl commented:
@jimingham Could you please review this?

adrian-prantl commented:
@swift-ci test

jrose-apple commented:
I'm really not sure the mangled name is the best place to get this information from. cc @rjmccall

adrian-prantl committed:
Represent Swift errors as DW_TAG_thrown_type in DWARF

This allows the debugger to display them alongside the function's
return value when finish-ing a function.

rdar://problem/29481673

adrian-prantl commented:
[Repurposing this pull request for a much cleaner alternative solution]

Represent Swift errors as DW_TAG_thrown_type in DWARF
    
This allows the debugger to display them alongside the function's
return value when finish-ing a function.

adrian-prantl commented:
@swift-ci test and merge

adrian-prantl commented:
Test failure appears to be unrelated.
@swift-ci smoke test and merge

jrose-apple commented:
I see the i386 test failing: https://ci.swift.org/job/swift-PR-osx/7095/consoleFull#1446795187fca400bf-2f4a-462e-b517-e058d770b2d7

```
17:24:54 
/Users/buildnode/jenkins/workspace/swift-PR-osx/branch-master/swift/test/DebugInfo/ErrorVar.swift:14:12: error: expected string not found in input
17:24:54  // CHECK: define {{.*}}void @_T06Errors6simpleys5Int64VKF(i64, %swift.refcounted* swiftself, %swift.error**)
17:24:54            ^
```

jrose-apple commented:
Ah, the mangled name is wrong. I'll get it for you.

jrose-apple commented:
Okay, even with that patch it's still failing.
